src/layers.jl

```diff
@@ -0,0 +1,133 @@
+"""
+PUSH_TO_LAYER
+
+A setinel to aid the creation of layers.
+If set to true, all the objects are pushed to the current layer.
+Can be accessed using PUSH_TO_LAYER[1]
+"""
+const PUSH_TO_LAYER = Array{Any,1}()
+# by default push all the objects to the video
+push!(PUSH_TO_LAYER, false)
+
+"""
+    @JLayer(frames, width, height, position, body)
+Calls the [`to_layer_m`](@ref) method to create a [`Layer`](@ref) out of the arguments given. 
+
+# Arguments
+- `frames`:a `UnitRange` that defines for which frames the layer is active
+- `width`: defines the width of the layer
+- `height`: defines the height of the layer
+- `position`: location of the center of the layer on the main canvas
+- `body`
+    - It contains all the objects(and thier respective actions) definitions for a layer
+    - A layer can have it's own separate background
+    - Anything defined within the `begin...end` block stays in the layer
+    - A Layer has it's own coordinate reference sysstem, i.e. it has it's own origin 
+    So eg : `Point(100, 100)` is different when defined in a layer and doesn't represent
+    the location 100, 100 on the main canvas 
+
+`width`, `height` and `position` are optional and default to the video's width, height and origin respectively.
+Layer declaration should take place before pushing objects to it if one is not using the macro
+
+# Example
+```julia
+function ground(args...)
+    background("white")
+    sethue("black")
+end
+
+function layer_ground(args...)
+    background("red")
+    sethue("white")
+end
+
+video = Video(500, 500)
+Background(1:100, ground)
+object((args...)->circle(O, 50, :fill))
+
+l1 = @Javis.Layer 10:70 100 100 Point(150, 150) begin
+    Background(10:70, layer_ground)
+    red_ball = Object(20:60, (args...)->object(O, "red"), Point(50,0))
+    act!(red_ball, Action(anim_rotate_around(2π, O)))
+end
+render(video; pathname="test.gif")
+```
+"""
+macro JLayer(frames, width, height, position, body)
+    esc(to_layer_m(frames, body, width = width, height = height, position = position))
+end
+
+macro JLayer(frames, body)
```

Comment:
Seems like ` @JLayer 1:100 begin` should work but I get a weird error:
```julia
 LoadError: BoundsError: attempt to access 0-element Vector{Video} at index [1]
│    Stacktrace:
│     [1] getindex
│       @ ./array.jl:801 [inlined]
│     [2] to_layer_m(frames::Expr, body::Expr)
│       @ Javis ~/Julia/Javis/src/layers.jl:80
│     [3] var"@JLayer"(__source__::LineNumberNode, __module__::Module, frames::Any, body::Any)
│       @ Javis ~/Julia/Javis/src/layers.jl:61
│     [4] lower
│       @ ./meta.jl:165 [inlined]
│     [5] methods_by_execution!(recurse::Any, methodinfo::Revise.CodeTrackingMethodInfo, docexprs::Dict{Module, Vector{Expr}}, mod::Module, ex::Expr; mode::Symbol, disablebp::Bool, always_rethrow::Bool, kwargs::Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
│       @ Revise ~/.julia/packages/Revise/1boD5/src/lowered.jl:181
```

Reply:
That should work now IG?